src/modules/utils/modals/gateaway/GateawaySignCtrl.js

```diff
@@ -0,0 +1,132 @@
+(function () {
+    'use strict';
+
+    /**
+     * @param {Base} Base
+     * @param {$rootScope.Scope} $scope
+     * @param {User} user
+     * @param {app.utils} utils
+     * @return {GateawaySignCtrl}
+     */
+    const controller = function (Base, $scope, user, utils) {
+
+        class GateawaySignCtrl extends Base {
+
+            /**
+             * @param {object} search
+             * @param {string} search.n Gateaway application name
+             * @param {string} search.r Gateaway referrer
+             * @param {string} search.i Gateaway icon path
+             * @param {string} search.d Gateaway data for sign
+             * @param {string} search.s Gateaway path for success auth
+             * @param {string} search.e Gateaway path for error auth
+             */
+            constructor(search) {
+                super($scope);
+                this.appName = search.n;
+                this.referrer = search.r;
+                this.iconPath = search.i;
+                this.data = search.d;
+                this.successPath = search.s;
+                this.errorPath = search.e;
+                this.hasIcon = false;
+
+                this._checkParams();
+
+                if (this.iconPath) {
+                    utils.loadImage(this.iconPath)
```

Comment:
First, check for a protocol. If it starts with `https://` load by absolute path, otherwise load by relative path.